compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -1598,9 +1599,18 @@ private <E extends BLangNode> E rewrite(E node, SymbolEnv env) {
     }
 
     private <E extends BLangNode> List<E> rewrite(List<E> nodeList, SymbolEnv env) {
-        for (int i = 0; i < nodeList.size(); i++) {
-            nodeList.set(i, rewrite(nodeList.get(i), env));
+        Queue<BLangSimpleVariableDef> previousQueue = this.annotationClosureReferences;
+        this.annotationClosureReferences = new LinkedList<>();
+        int size = nodeList.size();
+        for (int i = 0; i < size; i++) {
+            E node = rewrite(nodeList.remove(0), env);
+            Iterator<BLangSimpleVariableDef> iterator = annotationClosureReferences.iterator();
+            while (iterator.hasNext()) {
+                nodeList.add(rewrite((E) annotationClosureReferences.poll().var, env));
+            }
```

Comment:
Can't this be simplified by setting a new list/queue as `annotationClosureReferences` before every new relevant visit? 

Why does it have to be a queue?

Reply:
Do you mean without using 'previousQueue'? right?

Reply:
We may still need something like that (maybe not for module-level variables), just not here.